File: 07-interfaces/java/Pato.java
/*
 * Pato.java - hereda de UNA clase, firma DOS contratos
 *
 *   extends Ave                  <- lo que ES     (uno solo)
 *   implements Nadador, Volador  <- lo que SABE HACER (los que sean)
 */
public class Pato extends Ave implements Nadador, Volador {

    public Pato(String nombre, int edad) {
        super(nombre, edad, 3000);
    }

    @Override public String sonido() { return "Cuac"; }

    // --- contrato Nadador ---
    @Override public void nadar() {
        System.out.println("   " + nombre + " chapotea en la laguna");
    }
    @Override public double velocidadEnAgua() { return 8.0; }

    // --- contrato Volador ---
    @Override public void volar() {
        System.out.println("   " + nombre + " levanta vuelo");
    }
    @Override public int alturaMaxima() { return 1000; }
}

File: 07-interfaces/python/pato.py
"""
pato.py - hereda de UNA clase, firma DOS contratos
"""

from ave import Ave
from nadador import Nadador
from volador import Volador


class Pato(Ave, Nadador, Volador):

    def __init__(self, nombre, edad):
        super().__init__(nombre, edad, 3000)

    def sonido(self):
        return "Cuac"

    # --- contrato Nadador ---
    def nadar(self):
        print(f"   {self._nombre} chapotea en la laguna")

    def velocidad_en_agua(self):
        return 8.0

    # --- contrato Volador ---
    def volar(self):
        print(f"   {self._nombre} levanta vuelo")

    def altura_maxima(self):
        return 1000
